css/src/components/form/radio.scss

```diff
@@ -0,0 +1,117 @@
+$radio-circle-size: 1.25em !default;
+$radio-dot-size: 0.625em !default;
+$radio-dot-checked-color: $primary !default;
+$radio-dot-color: $text-subtle !default;
+$radio-timing-function: $input-timing-function !default;
+$radio-duration: $input-transition-duration !default;
+$radio-animation: boop !default;
+$radios-spacing: 0.5em !default;
+
+@mixin radio-dot {
+	position: absolute;
+	inset: 0;
+	width: $radio-dot-size;
+	height: $radio-dot-size;
+	margin: auto;
+	border-radius: 50%;
+	background-color: $body-background;
+	content: '';
+	overflow: hidden;
+}
+
+@mixin radio-dot-checked {
+	background: $radio-dot-checked-color;
+}
+
+.radio {
+	display: inline-flex;
+	position: relative;
+	align-items: flex-start;
+	line-height: 1.25;
+	cursor: pointer;
+
+	.radio-dot {
+		display: inline-block;
+		position: relative;
+		inset-block-start: 0.0625em;
+		flex-shrink: 0;
+		width: $radio-circle-size;
+		height: $radio-circle-size;
+		border: 1px solid $text;
+		border-radius: $radius-rounded;
+		color: $radio-dot-color;
+		appearance: none;
+
+		&::before {
+			@include radio-dot;
+		}
+	}
+
+	.radio-label-text {
+		margin: 0 $radios-spacing;
+	}
+
+	legend + & {
+		margin-block-start: $radios-spacing;
+	}
+
+	.radios & {
+		margin-top: $radios-spacing;
+	}
+
+	/* stylelint-disable selector-max-specificity */
+
+	&.is-hovered,
+	&:hover:not([disabled]) {
+		.radio-dot:not(:checked)::before {
+			@include radio-dot-checked;
+		}
+	}
+
+	/* stylelint-enable */
+
+	.radios.radios-vertical & {
+		display: flex;
+
+		+ .radio {
+			margin-top: $radios-spacing;
+			margin-left: 0;
+		}
+	}
+
+	/* stylelint-disable selector-no-qualifying-type */
+
+	input {
+		&.focus-visible,
+		&.is-focused {
+			@extend %focus;
+		}
+	}
+
+	input.is-checked,
+	input:checked,
+	.radio-dot.is-checked {
+		animation: $radio-animation $radio-duration $radio-timing-function 1;
+		border-color: $radio-dot-checked-color;
+
+		&::before {
+			@include radio-dot-checked;
+		}
+	}
+
+	@include reduceMotion {
```

Comment:
Personally, I think that this can be removed from this PR. The `boop` animation already falls under the reduced motion category IMO.

Reply:
That makes sense to me!

Reply:
reverted